### PR TITLE
Backport data type support for clip and mod

### PR DESCRIPTION
**Pull request by chinhuang007**
Backport data type support for clip and mod using the
auto-cast feature and add a test cast in test_node.py.

Signed-off-by: Chin Huang <chhuang@us.ibm.com>

### CODE REVIEW

File: onnx_tf/handlers/backend/mod.py [hunk 1]
```diff
@@ -1,38 +1,70 @@
...
-                "are not supported in Tensorflow.")
 class Mod(ArithmeticMixin, BackendHandler):
+  cast_map = {
+      tf.uint8: tf.int32,
+      tf.uint16: tf.int32,
+      tf.uint32: tf.int64,
+      tf.int8: tf.int32,
+      tf.int16: tf.int32
+  }
+  cast_map[tf.uint64] = tf.int64 if sys_config.auto_cast else None
```
Comment (winnietsang) [inline onnx_tf/handlers/backend/mod.py:20]: @chinhuang007 please move line 20 into args_check too.

**Review by winnietsang — CHANGES_REQUESTED** [review]

File: onnx_tf/handlers/backend/clip.py [hunk 2]
```diff
@@ -1,26 +1,37 @@
...
-@ps_description("Clip input in uint64 is not supported in Tensorflow.")
 class Clip(BackendHandler):
+  cast_map = {
+      tf.uint8: tf.int32,
+      tf.uint16: tf.int32,
+      tf.uint32: tf.int64,
+      tf.int8: tf.int32,
+      tf.int16: tf.int32
+  }
+  cast_map[tf.uint64] = tf.int64 if sys_config.auto_cast else None
```
Comment (winnietsang) [inline onnx_tf/handlers/backend/clip.py:19]: @chinhuang007 Please move line 19 into args_check. With current location, auto_cast will always be false and user can not change it at all.

Comment (chinhuang007) [reply onnx_tf/handlers/backend/clip.py]: Good catch! Fixed, please review again

**Review by winnietsang — APPROVED** [review]
LGTM

**Pull request merged**